src/common/utxobased/keymanager/cleaners.ts

```diff
@@ -0,0 +1,189 @@
+      }
+      const asCoinType = asOptional(asNumber, coinType)
+      return {
+        coinType: asCoinType(raw.coinType),
+        format: asOptionalPrivateKeyFormat(raw.format),
+        seed: asString(raw[`${coinName}Key`])
+      }
+    },
+    clean => {
+      const { coinType, format, seed } = clean
+      return {
+        coinType,
+        format,
+        [`${coinName}Key`]: seed
+      }
+    }
+  )
+}
+
+/**
+ * A cleaner for the public key format following the key-formats specification.
+ *
+ * (spec: https://github.com/EdgeApp/edge-core-js/blob/master/docs/key-formats.md)
+ */
+export interface PublicKey {
+  publicKeys: {
+    [format in CurrencyFormat]?: string
+  }
+}
+export const asPublicKey: Cleaner<PublicKey> = asObject({
+  publicKeys: asObject({
+    bip32: asOptional(asString),
+    bip44: asOptional(asString),
+    bip49: asOptional(asString),
+    bip84: asOptional(asString)
+  })
+})
+
+/**
+ * This utility returns a wallet's supported formats as specified in the
+ * key-formats specification.
+ */
+export const getSupportedFormats = (
+  format: CurrencyFormat
+): CurrencyFormat[] => {
+  switch (format) {
+    case 'bip32':
+      return ['bip32']
+    case 'bip44':
+      return ['bip44']
+    case 'bip49':
+      return ['bip49', 'bip84']
+    case 'bip84':
+      return ['bip49', 'bip84']
+    default:
+      throw new Error(`Unsupported format ${format}`)
+  }
+}
+
+/**
+ * A cleaner that desensitizes the walletInfo object, excluding sensitive
+ * keys (seed/mnemonic, sync key, data key, etc). By using this object type
+ * internally within the plugin, we can minimize risk.
+ *
+ * It also includes internal derived data (publicKey, format, supportedFormats, etc).
+ * This derived data is to be used internally within the plugin and saved to disk (publicKey).
+ */
+export interface NumbWalletInfo {
+  id: string
+  type: string
+  keys: {
+    format: CurrencyFormat
+    supportedFormats: CurrencyFormat[]
+    publicKey: PublicKey
+  }
+}
+export const asNumbWalletInfo = (
+  pluginInfo: PluginInfo
+): Cleaner<NumbWalletInfo> => {
+  return (walletInfo: EdgeWalletInfo): NumbWalletInfo => {
+    const { engineInfo, coinInfo } = pluginInfo
+    const { id, type } = walletInfo
+
+    const asCurrencyPrivateKey = asPrivateKey(coinInfo.name, coinInfo.coinType)
+
+    const publicKey = asMaybe(asPublicKey)(walletInfo.keys)
+    if (publicKey != null) {
+      const formats = Object.entries(publicKey.publicKeys)
+        // Filter out undefined values in the entries because cleaners allow
+        // undefined values for optional fields.
+        .filter(([, value]) => value != null)
+        // Map to the entry's key as a currency format
+        .map(([format]) => asMaybe(asCurrencyFormat)(format))
+        // Filter out any formats that didn't pass the cleaner assertion
+        .filter(
+          (format?: CurrencyFormat): format is CurrencyFormat => format != null
+        )
```

Comment:
This had to be fixed. The formats needed to be changed from accessing the `Object.keys(publicKey.publicKeys)` to `Object.entries(publicKey.publicKeys)` so that way we can filter out the values (because they may be undefined).